notes/28_06_2024_tb_legacy_boot.md

```diff
@@ -0,0 +1,135 @@
+  3  : 3a3f780f11a4b49969fcaa80cd6e3957c33b2275
+  4  : 29e503140459f9ae7b6fb502ab0486d941de948c
+  5  : e39b032eaa782999471de86e00fcb3c7c254eab6
+  6  : 3a3f780f11a4b49969fcaa80cd6e3957c33b2275
+  7  : 3a3f780f11a4b49969fcaa80cd6e3957c33b2275
+  8  : 0000000000000000000000000000000000000000
+  9  : 0000000000000000000000000000000000000000
+  10 : 0000000000000000000000000000000000000000
+  11 : 0000000000000000000000000000000000000000
+  12 : 0000000000000000000000000000000000000000
+  13 : 0000000000000000000000000000000000000000
+  14 : 0000000000000000000000000000000000000000
+  15 : 0000000000000000000000000000000000000000
+  16 : 0000000000000000000000000000000000000000
+  17 : b8e108dfa488195c2bd86a2cba16e983e98e37ed
+  18 : 0000000000000000000000000000000000000000
+  19 : 0000000000000000000000000000000000000000
+  20 : 0000000000000000000000000000000000000000
+  21 : 0000000000000000000000000000000000000000
+  22 : 0000000000000000000000000000000000000000
+  23 : 0000000000000000000000000000000000000000
+sha256 :
+  0  : e21b703ee69c77476bccb43ec0336a9a1b2914b378944f7b00a10214ca8fea93
+  1  : 37a60d915ea369df3b4c24ce34c8ee115d2a87f8362b64d982941ba6ca5d3f9b
+  2  : 8c92c8075ba25c46cb3424502a41184b2cf8994aa6f32b24c266cf45bfb36735
+  3  : e21b703ee69c77476bccb43ec0336a9a1b2914b378944f7b00a10214ca8fea93
+  4  : 40f1272fd32ae0ba5c2000bc600199b0a4422a359b100033140120359690d717
+  5  : e778a0f4176fdba0964f1b709aa41fed0926accce6be1684fcf83e7c04dbb823
+  6  : e21b703ee69c77476bccb43ec0336a9a1b2914b378944f7b00a10214ca8fea93
+  7  : e21b703ee69c77476bccb43ec0336a9a1b2914b378944f7b00a10214ca8fea93
+  8  : 0000000000000000000000000000000000000000000000000000000000000000
+  9  : 0000000000000000000000000000000000000000000000000000000000000000
+  10 : 0000000000000000000000000000000000000000000000000000000000000000
+  11 : 0000000000000000000000000000000000000000000000000000000000000000
+  12 : 0000000000000000000000000000000000000000000000000000000000000000
+  13 : 0000000000000000000000000000000000000000000000000000000000000000
+  14 : 0000000000000000000000000000000000000000000000000000000000000000
+  15 : 0000000000000000000000000000000000000000000000000000000000000000
+  16 : 0000000000000000000000000000000000000000000000000000000000000000
+  17 : a67681a54c9e46ae39316d7cb2c983d052f8190478b916224420dd5a719d358f
+  18 : 7fd9068bb68527a096b44bf6831b8ad0cbb8b53e045bfd3ecc427bd68ec27de3
+  19 : 0000000000000000000000000000000000000000000000000000000000000000
+  20 : 0000000000000000000000000000000000000000000000000000000000000000
+  21 : 0000000000000000000000000000000000000000000000000000000000000000
+  22 : 0000000000000000000000000000000000000000000000000000000000000000
+  23 : 0000000000000000000000000000000000000000000000000000000000000000
+```
+
+SHA256 hash of PCR17 mathes to the pre-calculated one.
+SHA1 does not.
+In the [blog post](https://blog.3mdeb.com/2020/2020-08-13-trenchboot-event-log/)
+both of them used to match.
```

Comment:
Likely related to the use of a modern toolchain and https://github.com/TrenchBoot/secure-kernel-loader/commit/c00c554c793508a5719661105a6d6929249b0945.  IIRC, the issue manifested itself for SHA1 only.

Reply:
Which toolchain? The one to build SKL/LZ?

Reply:
Yes, although it's a yocto build, so maybe LZ was built manually for the test from the blog post.

Reply:
Doesn't look like that is the issue. In the log, there is only one `shasum calculated:` and it looks like SHA256. This is how the code should work in https://github.com/TrenchBoot/landing-zone/blob/4dc904bf70d6913a5706bdfc6f7781a9ff039d38/main.c#L155, later it was improved to include SHA1 for TPM2 as well: https://github.com/TrenchBoot/landing-zone/blob/master/main.c#L162.

This was fixed on main branch on Jul 15, 2021, and blog post was released Aug 13, 2020. It was dormant on https://github.com/TrenchBoot/landing-zone/pull/52 before being rebased as https://github.com/TrenchBoot/landing-zone/pull/65 one year later. `meta-trenchboot` switched to our fork of LZ after the post (Nov 4, 2020 https://gitlab.com/trenchboot1/3mdeb/meta-trenchboot/-/commit/aa97eee114ed08c94bb71a4bf9efd75d85d43286), so it must've been manual build (less likely: time travel).

Reply:
7febf807068f1fbb77dc14462f9696810998f16e